### ISSUE TITLE
Feature Request — Warn user of possible wallet update needed

### ISSUE THREAD

**Issue by gabrielnergaard** [Veil-Project/veil#404]
Knowing about a wallet update is always convenient to users, and in some cases of critical importance.

To avoid external dependencies, like checking GitHub, we could take the approach used by Dash. Whenever the wallet detects that peers on the network are using a later version, an impossible-not-to-see yellow bar is present on the wallet overview screen stating something like:

"We've noticed a majority of peers on the network running a later version of the wallet than you. You may want to check to see if there's an update available. For reference you are running 0.13.0 and the others are running 0.13.1"

They don't link to GitHub, just simply provide that advice to the user.

Can we do this in the Veil wallet?

**Comment by jakimanboy** [comment]
I support this idea as it doesn't create any extra polling over the network i.e. to our GitHub, and merely compares the connected peer's wallet version to see if there is a discrepancy. So this would be an effective alert to notify that the user may be running an outdated version. 

As for the text, instead of it displaying directly, maybe the GUI could just display a prominent blinking RED icon in the bottom status bar with the warning text only showing if you hover over the icon with the mouse cursor.

**Comment by gabrielnergaard** [comment]
If there’s room for the text it would be better, even if aesthetically less pleasing, because the goal here is that the user updates. Showing the text will result in that happening more often, given that some users won’t know to hover over a blinking RED icon. (On the Mac, for example, critical information only available on hover is definitely not a UI pattern recommended by Apple.)

**Comment by jakimanboy** [comment]
Makes sense. I guess it can display a colored warning text bar in a similar manner like the one that shows up at the top of the wallet when you compile your own wallet which tells you that it's a pre-release build test build. Considering that is already something the current wallet is capable of, it would be good to make use of it for other warning types like this one.

**Comment by gabrielnergaard** [comment]
If the current wallet already contains a place/mechanism to provide messaging to the user, then that’d be perfect.

**Comment by furszy** [comment]
NACK, please read about the bitcoin alert system, vulnerabilities and why it was deprecated on 0.13.

**Comment by gabrielnergaard** [comment]
I just googled this, and the wikipedia page on the bitcoin alert system says:

"In 2016, the alert system was retired[1] because of the possibility of privileged users sending political alert messages, and because of the possibility of the alert key having been taken from Mark Karpelès by the Japanese police in 2014.[3]”

Is this the vulnerability you’re concerned about here?

**Comment by JSKitty** [comment]
I have to agree with the Feature Request~ In the original post, Gabriel mentioned a majority of users needing to be detected running on a newer version and/or protocol. This **doesn't** require the original Bitcoin alert system to function. It could be a simple notification written from scratch. No centralized polling, no global alert system (again, avoiding a central authority issuing an update), but the nodes themselves detect if the network has a majority of a higher version, and alerts the user locally if suitable. To avoid mis-fires it could require a minimum amount of peers before polling the peers list for 50% update calcs. E.G: A minimum of 10 peers before your node begins to poll peers for versions.

**Comment by gabrielnergaard** [comment]
This is exactly what I proposed in the original request. This is exactly what @jakimanboy who is a team leader in Veil said he agreed with.

Please, @furszy, explain yourself?

**Comment by gabrielnergaard** [comment]
@furszy That probably came across wrong. I'm not asking you to "explain yourself". What I mean is this — the Veil team members, including the named Chief Administrative Officer, has asked for a feature, which would be a great convenience to the users, and sometimes of critical importance to the users, and would keep the product decentralized, as all it depends on is an analysis of its peers on the network.

You responded with a NACK, and a justification that is irrelevant to the feature being requested.

What I want is your opinion of how products get created in a project like Veil? Are you the ultimate arbiter and gatekeeper of what gets implemented in this product? Does the desires and opinions of the rest of the team and the community matter?

CC @4x13 @presstab

**Comment by veilgets** [comment]
Bounty: 437.50

**Comment by JSKitty** [comment]
Oh, sweet.

**Comment by veilgets** [comment]
Developer: JSKitty

**Comment by CaveSpectre11** [comment]
Adjusted Bounty: 1000.00

**Comment by MatWaller** [comment]
![here](https://user-images.githubusercontent.com/8047888/70414479-1af31480-1a52-11ea-8e96-858342cd369a.png)
Could perhaps use the UI that exists for warning a user when a test build is being used (with a different message of course) @JSKitty

**Comment by JSKitty** [comment]
> ![here](https://user-images.githubusercontent.com/8047888/70414479-1af31480-1a52-11ea-8e96-858342cd369a.png)
> Could perhaps use the notification that exists for warning a user when a test build is being used (with a different message of course) @JSKitty

That's exactly how it works already, can see the latest code in my last PR using the warning utils.
